src/configuration/vimrc.ts

```diff
@@ -42,33 +42,54 @@ class VimrcImpl {
    * Adds a remapping from .vimrc to the given configuration
    */
   private static addRemapToConfig(config: IConfiguration, remap: IVimrcKeyRemapping): void {
-    const remaps = (() => {
+    const mappings = (() => {
       switch (remap.keyRemappingType) {
+        case 'map':
+          return [
+            config.normalModeKeyBindings,
+            config.visualModeKeyBindings,
+            config.insertModeKeyBindings,
+            config.commandLineModeKeyBindings,
+          ];
         case 'nmap':
-          return config.normalModeKeyBindings;
+          return [config.normalModeKeyBindings];
         case 'vmap':
-          return config.visualModeKeyBindings;
+          return [config.visualModeKeyBindings];
         case 'imap':
-          return config.insertModeKeyBindings;
+          return [config.insertModeKeyBindings];
         case 'cmap':
-          return config.commandLineModeKeyBindings;
+          return [config.commandLineModeKeyBindings];
+        case 'noremap':
+          return [
+            config.normalModeKeyBindingsNonRecursive,
+            config.visualModeKeyBindingsNonRecursive,
+            config.insertModeKeyBindingsNonRecursive,
+            config.commandLineModeKeyBindingsNonRecursive,
+          ];
         case 'nnoremap':
-          return config.normalModeKeyBindingsNonRecursive;
+          return [config.normalModeKeyBindingsNonRecursive];
         case 'vnoremap':
-          return config.visualModeKeyBindingsNonRecursive;
+          return [config.visualModeKeyBindingsNonRecursive];
         case 'inoremap':
-          return config.insertModeKeyBindingsNonRecursive;
+          return [config.insertModeKeyBindingsNonRecursive];
         case 'cnoremap':
-          return config.commandLineModeKeyBindingsNonRecursive;
+          return [config.commandLineModeKeyBindingsNonRecursive];
         default:
+          console.warn('Encountered an unrecognized mapping type.');
           return undefined;
       }
     })();
 
-    // Don't override a mapping present in settings.json; those are more specific to VSCodeVim.
-    if (remaps && !remaps.some(r => _.isEqual(r.before, remap!.keyRemapping.before))) {
-      remaps.push(remap.keyRemapping);
+    if (!mappings) {
```

Comment:
This shouldn't be necessary - the `forEach` will be a no-op if `mappings` is empty

Reply:
`mappings` could be undefined (not empty) from switch statement.  Also tsc complains that mappings might be null or undefined.  So I think this is necessary to keep the error of an unrecognized mapping type as non-fatal (as it was before).

Reply:
Oh I see. Could you remove this check and do `mappings?.forEach` instead?

Reply:
Done.